network/msgqueue/indexes.go

```diff
@@ -7,31 +7,60 @@ import (
 )
 
 // IBFTRoundIndexKey is the ibft index key
-func IBFTRoundIndexKey(lambda []byte, round uint64, validatorPk []byte) string {
-	return fmt.Sprintf("lambda_%s_round_%d_pubKey_%s",
-		hex.EncodeToString(lambda), round, hex.EncodeToString(validatorPk))
+func IBFTRoundIndexKey(lambda []byte, seqNumber uint64, round uint64) string {
+	return fmt.Sprintf("lambda_%s_seqNumber_%d_round_%d", hex.EncodeToString(lambda), seqNumber, round)
 }
 
 func iBFTMessageIndex() IndexFunc {
 	return func(msg *network.Message) []string {
 		if msg.Type == network.NetworkMsg_IBFTType {
 			return []string{
-				IBFTRoundIndexKey(msg.Lambda, msg.SignedMessage.Message.Round, msg.SignedMessage.Message.ValidatorPk),
+				IBFTRoundIndexKey(msg.Lambda, msg.SignedMessage.Message.SeqNumber, msg.SignedMessage.Message.Round),
 			}
 		}
 		return []string{}
 	}
 }
 
 // SigRoundIndexKey is the SSV node signature collection index key
-func SigRoundIndexKey(lambda []byte) string {
-	return fmt.Sprintf("sig_lambda_%s", hex.EncodeToString(lambda))
+func SigRoundIndexKey(lambda []byte, seqNumber uint64) string {
+	return fmt.Sprintf("sig_lambda_%s_seqNumber_%d", hex.EncodeToString(lambda), seqNumber)
 }
 func sigMessageIndex() IndexFunc {
 	return func(msg *network.Message) []string {
 		if msg.Type == network.NetworkMsg_SignatureType {
 			return []string{
-				SigRoundIndexKey(msg.Lambda),
+				SigRoundIndexKey(msg.Lambda, msg.SignedMessage.Message.SeqNumber),
+			}
+		}
+		return []string{}
+	}
+}
+
+// DecidedIndexKey is the ibft decisions index key
+func DecidedIndexKey(lambda []byte) string {
+	return fmt.Sprintf("decided_lambda_%s", hex.EncodeToString(lambda))
+}
+func decidedMessageIndex() IndexFunc {
+	return func(msg *network.Message) []string {
+		if msg.Type == network.NetworkMsg_DecidedType {
+			return []string{
+				DecidedIndexKey(msg.Lambda),
+			}
+		}
+		return []string{}
+	}
+}
+
+// SyncIndexKey is the ibft sync index key
+func SyncIndexKey(pubKey []byte) string {
+	return fmt.Sprintf("sync_publicKey_%s", hex.EncodeToString(pubKey))
```

Comment:
@nivBlox should receive identifier not public key

Reply:
identifier is built from pubkey and role. sync msg's not familiar with role

Reply:
every message has an identifier.. we either identify msgs by identifier or something else but let's not mix

Reply:
each msg type has differ index you cant force the same for all.. 
also sync is at validator level that doesnt have identifier